npm_and_yarn/spec/dependabot/npm_and_yarn/file_updater/pnpm_workspace_updater_spec.rb

```diff
@@ -0,0 +1,321 @@
+            }],
+            previous_requirements: [{
+              file: "pnpm-workspace.yaml",
+              requirement: "^3.3.0",
+              groups: [],
+              source: nil
+            }]
+          ),
+          Dependabot::Dependency.new(
+            name: "left-pad",
+            version: "1.0.1",
+            package_manager: "npm_and_yarn",
+            requirements: [{
+              file: "pnpm-workspace.yaml",
+              requirement: "^1.0.3",
+              groups: [],
+              source: nil
+            }],
+            previous_requirements: [{
+              file: "pnpm-workspace.yaml",
+              requirement: "^1.0.1",
+              groups: [],
+              source: nil
+            }]
+          )
+        ]
+      end
+
+      its(:content) { is_expected.to include "prettier: ^3.3.3" }
+      its(:content) { is_expected.to include "left-pad: ^1.0.3" }
+    end
+
+    context("with catalog group dependency") do
+      let(:project_name) { "pnpm/catalogs_react" }
+      let(:dependencies) do
+        [
+          Dependabot::Dependency.new(
+            name: "react",
+            version: "18.0.0",
+            package_manager: "npm_and_yarn",
+            requirements: [{
+              file: "pnpm-workspace.yaml",
+              requirement: "^18.2.3",
+              groups: [],
+              source: nil
+            }],
+            previous_requirements: [{
+              file: "pnpm-workspace.yaml",
+              requirement: "^18.0.0",
+              groups: [],
+              source: nil
+            }]
+          ),
+          Dependabot::Dependency.new(
+            name: "react-dom",
+            version: "18.0.0",
+            package_manager: "npm_and_yarn",
+            requirements: [{
+              file: "pnpm-workspace.yaml",
+              requirement: "^18.2.3",
+              groups: [],
+              source: nil
+            }],
+            previous_requirements: [{
+              file: "pnpm-workspace.yaml",
+              requirement: "^18.0.0",
+              groups: [],
+              source: nil
+            }]
+          )
+        ]
+      end
+
+      its(:content) { is_expected.to include "react: ^18.2.3" }
+      its(:content) { is_expected.to include "react-dom: ^18.2.3" }
+    end
+
+    context("with catalog group dependency") do
+      let(:project_name) { "pnpm/catalogs_multiple_reacts" }
+      let(:dependencies) do
+        [
+          Dependabot::Dependency.new(
+            name: "react",
+            version: "18.0.0",
+            package_manager: "npm_and_yarn",
+            requirements: [{
+              file: "pnpm-workspace.yaml",
+              requirement: "^18.2.3",
+              groups: [],
+              source: nil
+            }],
+            previous_requirements: [{
+              file: "pnpm-workspace.yaml",
+              requirement: "^18.0.0",
+              groups: [],
+              source: nil
+            }]
+          ),
```

Comment:
nit, 

Maybe we should extract this repeated logic into a method so we can quickly see the pertinent info, eg

```ruby
create_dependency(name: "react",
                  version: "18.0.0",
                  required_version: "^18.2.3",
                  previous_required_version: "^18.0.0")
```